### PR TITLE
feat(CCM): update certificate export

### PR DESCRIPTION
**Pull request by zhangtao505**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
feat(CCM): update certificate export

**Special notes for your reviewer**:

**Release note**:

 ./scripts/codecheck.sh ./huaweicloud/services/ccm

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        4      1749      138        23     1588        261            66.97
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~ccm/resource_huaweicloud_ccm_private_ca.go       741       58        11      672        117            17.41
~source_huaweicloud_ccm_certificate_push.go       344       29         5      310         75            24.19
~rce_huaweicloud_ccm_private_certificate.go       484       41         4      439         43             9.79
~weicloud_ccm_private_certificate_export.go       180       10         3      167         26            15.57
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     4      1749      138        23     1588        261            66.97
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 53704 bytes, 0.054 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
20 ccm resourcePrivateCACreate huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go:211:1
14 ccm resourceCcmCertificateRead huaweicloud/services/ccm/resource_huaweicloud_ccm_certificate_push.go:170:1
14 ccm ccmPrivateCertificateExport huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go:92:1
13 ccm rmPushedCert huaweicloud/services/ccm/resource_huaweicloud_ccm_certificate_push.go:301:1
12 ccm resourcePrivateCADelete huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go:537:1
8 ccm resourcePrivateCARead huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go:435:1
7 ccm resourceCcmPrivateCertificateRead huaweicloud/services/ccm/resource_huaweicloud_ccm_private_certificate.go:384:1
7 ccm resourceCcmPrivateCertificateUpdate huaweicloud/services/ccm/resource_huaweicloud_ccm_private_certificate.go:349:1
7 ccm resourcePrivateCAUpdate huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go:637:1
7 ccm rePushOrDelete huaweicloud/services/ccm/resource_huaweicloud_ccm_certificate_push.go:244:1
Average: 4.11

==> Checking for golangci-lint...

==> Checking for Nolint directives...

==> Checking for TF features in ccm...
  -> the resource in resource_huaweicloud_ccm_certificate_push.go should can be imported

==> Checking for misspell in ccm...

==> Checking for code complexity in ./huaweicloud/services/acceptance/ccm...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        4       765       60        10      695         22            13.29
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~e_huaweicloud_ccm_certifiacte_push_test.go       140       13         0      127          8             6.30
~esource_huaweicloud_ccm_private_ca_test.go       293       19         4      270          8             2.96
~uaweicloud_ccm_private_certificate_test.go       168       17         2      149          6             4.03
~oud_ccm_private_certificate_export_test.go       164       11         4      149          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     4       765       60        10      695         22            13.29
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 23571 bytes, 0.024 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
5 ccm getPrivateCAResourceFunc huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:18:1
5 ccm getCertificatePushResourceFunc huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_certifiacte_push_test.go:22:1
4 ccm getCertificateResourceFunc huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_certificate_test.go:18:1
1 ccm tesCmdbCertificate_tagsUpdate huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_certificate_test.go:143:1
1 ccm tesCmdbCertificate_basic huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_certificate_test.go:115:1
Average: 1.52

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/ccm...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/ccm...
./huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go:85:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go:96:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go:107:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go:118:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:106:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:135:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:165:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_ca_test.go:244:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_certificate_test.go:114:// lintignore:AT004
./huaweicloud/services/acceptance/ccm/resource_huaweicloud_ccm_private_certificate_test.go:142:// lintignore:AT004

==> Cleanup patch...

Check Completed!

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

 make testacc TEST=./huaweicloud/services/acceptance/ccm TESTARGS='-run TestAccCcmPrivateCertificateExport_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ccm -v -run TestAccCcmPrivateCertificateExport_basic -timeout 360m -parallel 4
=== RUN   TestAccCcmPrivateCertificateExport_basic
=== PAUSE TestAccCcmPrivateCertificateExport_basic
=== CONT  TestAccCcmPrivateCertificateExport_basic
--- PASS: TestAccCcmPrivateCertificateExport_basic (47.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ccm       47.925s

### CODE REVIEW

File: docs/data-sources/ccm_private_certificate_export.md [hunk 1]
```diff
@@ -61,4 +65,9 @@ In addition to all arguments above, the following attributes are exported:
   
 * `signed_and_enveloped_data` - Indicates the National Security GMT0010 Standard Specification for
   Encrypting Private Keys - Signature Digital Envelope.
-  
\ No newline at end of file
+
+* `keystore_pass` - Indicates the keystore password. If argument "password" passed in, it will be empty.
+
+* `server_pfx` - Indicates the certificate file with certificate chain file. Encoding by base64.
```
Comment (ShiChangkuo) [inline docs/data-sources/ccm_private_certificate_export.md]: the description of this parameter is not correct.

File: huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go [hunk 2]
```diff
@@ -33,6 +33,20 @@ func TestAccCcmPrivateCertificateExport_basic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(expEesourceName, "certificate_chain"),
 				),
 			},
+			{
+				Config: testAccCcmPrivateCertificateExport_TOMCAT(rName),
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go]: it's better to add a new acceptance test to check tomcat and iis type.

File: huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go [hunk 3]
```diff
@@ -71,14 +85,36 @@ func TestAccCcmPrivateCertificateExport_SM2(t *testing.T) {
...
     certificate_id = huaweicloud_ccm_private_certificate.test.id
 }`, tesCmdbCertificate_basic(commonName))
 }
 
+// lintignore:AT004
+func testAccCcmPrivateCertificateExport_TOMCAT(commonName string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_ccm_private_certificate_export" "test5" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go]: it's better to change `test3`, `test5` and `test6` to more meaningful name.

File: huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go [hunk 4]
```diff
@@ -82,44 +98,83 @@ func ccmPrivateCertificateExport(_ context.Context, d *schema.ResourceData, meta
...
 	}
 	return nil
 }
 
-func buildExpCertificateBodyParams(d *schema.ResourceData) map[string]interface{} {
+func readAndSave(f *zip.File, d *schema.ResourceData) {
+	fileName := f.Name
+	inFile, err := f.Open()
+	if err != nil {
+		log.Printf("[WARN] error unziping certificate file: %s", fileName)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go]: should return when we encounter an error.

File: huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go [hunk 5]
```diff
@@ -82,44 +98,83 @@ func ccmPrivateCertificateExport(_ context.Context, d *schema.ResourceData, meta
...
-	)
-	if err := mErr.ErrorOrNil(); err != nil {
-		return diag.Errorf("error setting CCM private certificate export fields: %s", err)
+	if isCompressed {
+		// save certificate file as attr
+		data, err := io.ReadAll(expCertificateResp.Body)
+		if err != nil {
+			return diag.Errorf("error reading zip certificate: %s", certId)
+		}
+		zipReader, _ := zip.NewReader(bytes.NewReader(data), int64(len(data)))
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go]: should check the error

File: huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go [hunk 6]
```diff
@@ -82,44 +98,83 @@ func ccmPrivateCertificateExport(_ context.Context, d *schema.ResourceData, meta
...
+func readAndSave(f *zip.File, d *schema.ResourceData) {
+	fileName := f.Name
+	inFile, err := f.Open()
+	if err != nil {
+		log.Printf("[WARN] error unziping certificate file: %s", fileName)
+	}
+	defer inFile.Close()
+	content, _ := io.ReadAll(inFile)
+	if fileName == "keystorePass.txt" {
+		d.Set("keystore_pass", string(content))
```
Comment (ShiChangkuo) [inline huaweicloud/services/ccm/data_source_huaweicloud_ccm_private_certificate_export.go]: it's better to use `multierror` to wrap the result

File: docs/data-sources/ccm_private_certificate_export.md (hunk 1 above)
Comment (zhangtao505) [reply docs/data-sources/ccm_private_certificate_export.md]: down

File: huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go [hunk 7]
```diff
@@ -10,20 +10,12 @@ import (
...
-	var obj interface{}
 	rName := acceptance.RandomAccResourceName()
-	expEesourceName := "data.huaweicloud_ccm_private_certificate_export.test3"
-	resourceName := "huaweicloud_ccm_private_certificate.test"
-	rc := acceptance.InitResourceCheck(
-		resourceName,
-		&obj,
-		getCertificateResourceFunc,
-	)
+	expEesourceName := "data.huaweicloud_ccm_private_certificate_export.basiccert"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ccm/data_source_huaweicloud_ccm_private_certificate_export_test.go]:
```suggestion
	dataSourceName := "data.huaweicloud_ccm_private_certificate_export.basiccert"
	dc := acceptance.InitDataSourceCheck(dataSourceName )
```

**Pull request merged**